pkg/host/internal/infiniband/infiniband_test.go

```diff
@@ -0,0 +1,47 @@
+package infiniband
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	netlinkLibPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink"
+	netlinkMockPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink/mock"
+	hostMockPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/mock"
+)
+
+var _ = Describe("infiniband interface implementation", func() {
+	It("should create infiniband helper if guid config path is empty", func() {
+		testCtrl := gomock.NewController(GinkgoT())
+		netlinkLibMock := netlinkMockPkg.NewMockNetlinkLib(testCtrl)
+		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil).Times(1)
+
+		_, err := New(netlinkLibMock, nil, nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("should assign guids if guid pool is nil", func() {
+		testCtrl := gomock.NewController(GinkgoT())
+		netlinkLibMock := netlinkMockPkg.NewMockNetlinkLib(testCtrl)
+		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil).Times(1)
+		netlinkLibMock.EXPECT().LinkSetVfNodeGUID(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(link netlinkLibPkg.Link, vf int, nodeguid net.HardwareAddr) error {
+			// Return the GUID in an error message as a feedback mechanism
+			return fmt.Errorf(nodeguid.String())
+		}).Times(1)
+		netlinkLibMock.EXPECT().LinkSetVfPortGUID(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+		kernelHelperMock := hostMockPkg.NewMockHostManagerInterface(testCtrl)
+		kernelHelperMock.EXPECT().Unbind(gomock.Any()).Return(nil).Times(1)
+
+		ib, err := New(netlinkLibMock, nil, nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = ib.ConfigureVfGUID("randomAddr", "randomAddr", 0, nil)
+		// A mocked function called inside ib.ConfigureVfGUID should return an error with a GUID assigned by ib interface as an error
+		// We then validate the error message to check that it contains a valid GUID
+		Expect(err).To(HaveOccurred())
+		_, err = ParseGUID(err.Error())
+		Expect(err).NotTo(HaveOccurred())
+	})
+})
```

Comment:
```suggestion
var _ = Describe("infiniband interface implementation", func() {
	var (
		testCtrl       *gomock.Controller
		netlinkLibMock *netlinkMockPkg.MockNetlinkLib
		hostMock       *hostMockPkg.MockHostManagerInterface
	)
	BeforeEach(func() {
		testCtrl = gomock.NewController(GinkgoT())
		netlinkLibMock = netlinkMockPkg.NewMockNetlinkLib(testCtrl)
		hostMock = hostMockPkg.NewMockHostManagerInterface(testCtrl)
	})

	AfterEach(func() {
		testCtrl.Finish()
	})

	It("should create infiniband helper if guid config path is empty", func() {
		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil)
		_, err := New(netlinkLibMock, hostMock, hostMock)
		Expect(err).NotTo(HaveOccurred())
	})
	It("should assign guids if guid pool is nil", func() {
		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil)
		var generatedGUID string
		pfLinkMock := netlinkMockPkg.NewMockLink(testCtrl)
		netlinkLibMock.EXPECT().LinkSetVfNodeGUID(pfLinkMock, 0, gomock.Any()).DoAndReturn(
			func(link netlinkLibPkg.Link, vf int, nodeguid net.HardwareAddr) error {
				// save generated GUID to validate that it is valid
				generatedGUID = nodeguid.String()
				return nil
			})
		netlinkLibMock.EXPECT().LinkSetVfPortGUID(pfLinkMock, 0, gomock.Any()).Return(nil)
		hostMock.EXPECT().Unbind("0000:d8:00.2").Return(nil)

		ib, err := New(netlinkLibMock, hostMock, hostMock)
		Expect(err).NotTo(HaveOccurred())

		err = ib.ConfigureVfGUID("0000:d8:00.2", "0000:d8:00.0", 0, pfLinkMock)
		Expect(err).NotTo(HaveOccurred())
		// validate that generated GUID is valid
		_, err = ParseGUID(generatedGUID)
		Expect(err).NotTo(HaveOccurred())
	})
})
```
For me this is a bit confusing, I propose to use a different technique +  let's add validation for mock calls (check proposed changes).